src/seabolt/src/bolt/v3.c

```diff
@@ -882,9 +848,24 @@ void BoltProtocolV3_extract_metadata(struct BoltConnection* connection, struct B
                 struct BoltValue* value = BoltDictionary_value(metadata, i);
                 switch (BoltValue_type(value)) {
                 case BOLT_STRING: {
-                    char new_connection_id[MAX_CONNECTION_ID_SIZE];
-                    strncpy(new_connection_id, BoltString_get(value), (size_t) value->size+1);
-                    const char* old_connection_id = BoltConnection_id(connection);
+                    const char* old_connection_id_ptr = BoltConnection_id(connection);
```

Comment:
Would it be simpler if we do something like this?

```
char old_connection_id[MAX_CONNECTION_ID_SIZE];
strcpy(old_connection_id, BoltConnection_id(connection));

if (snprintf(state->connection_id, MAX_CONNECTION_ID_SIZE, "%s, %s", old_connection_id, new_connection_id) >= MAX_CONNECTION_ID_SIZE) {
    state->connection_id[MAX_CONNECTION_ID_SIZE-1] = '\0';
    BoltLog_error(.....);
}
```

Reply:
It would indeed :) So I used your approach above but the implementation wasn't functionally identical to what I had before, so I modified it for when you encounter a new `connection_id` that would overrun the buffer, instead of truncating it, leave the original `connection_id` in place.

I'm building/testing on Arch Linux, and things are working out fine like this:

```shell
$ OPENSSL_ROOT_DIR=~/openssl/usr/local/lib/ ./make_debug.sh
$ NEOCTRL_LOCAL_PACKAGE=~/Downloads/neo4j-community-3.5.0-rc01-unix.tar.gz ./run_tests.sh 
```

As you can see, I had to use a local build of openssl (v1.1.2) because the arch linux package doesn't include the static libraries, but none of these changes touch that.

I'm going to see if I can get the docker CI stuff to work without TeamCity using a container running neo4j:latest to see if I can reproduce some of these issues.

Reply:
I made a few modifications to the run_tests.sh script so that it could easily be adapted to running the test-suite against an existing server (locally, or a docker service container ala docker-compose). If I start the server myself (either the one that boltkit configured or the docker container neo4j:latest), and run against that from a regular terminal on my machine, everything is happy as usual:

```shell
$ neoctrl-start build/server/neo4j-community-3.5.0-rc01/
http://127.0.0.1:7474
bolt://127.0.0.1:7687

$ HAVE_SERVER=yes ./run_tests.sh 
Seabolt test run started at 2018-11-30T17:58:51
sleep: missing operand
Try 'sleep --help' for more information.
-- Checking server
-- Attempt #1
[connector]: Version 1.7.1-dev [e6ea1aa]
[pool-1]: Creating pool towards localhost:7687
[pool-1]: Acquiring connection from the pool towards localhost:7687
[addr]: Resolving address localhost:7687
[addr]: Host resolved to 2 IP addresses
[conn-1]: Opening IPv6 connection to ::1 at port 7687
[conn-1]: Securing socket
[pool-1]: Openssl reported error 'self signed certificate' with code '18' when establishing trust, but resuming handshake since trust verification is set to be skipped
[pool-1]: Openssl established trust
[conn-1]: Performing handshake
[conn-1]: (Sent 20 of 20 bytes)
[conn-1]: Received 4 of 4 bytes
[conn-1]: <SET protocol_version=3>
[conn-1]: Remote endpoint is ::1:7687
[conn-1]: Local endpoint is ::1:39728
[conn-1]: <CONNECTED>
[conn-1]: Initialising connection
[conn-1]: C[0] HELLO [{scheme: basic, principal: neo4j, credentials: ********, user_agent: seabolt/1.7.1-dev}]
[conn-1]: (Sent 87 of 87 bytes)
[conn-1]: Received 53 of 2..8192 bytes
[conn-1]: S[0] SUCCESS [{server: Neo4j/3.5.0-rc01, connection_id: bolt-1}]
[conn-1]: <SET server="Neo4j/3.5.0-rc01">
[conn-1]: <SET connection_id="bolt-1">
[conn-1, bolt-1]: <READY>
[conn-1, bolt-1]: C[1] BEGIN [{}]
[conn-1, bolt-1]: C[2] RUN [UNWIND range(1, 10000) AS n RETURN n, {}, {}]
[conn-1, bolt-1]: C[3] PULL_ALL []
[conn-1, bolt-1]: C[4] COMMIT []
[conn-1, bolt-1]: (Sent 65 of 65 bytes)
[conn-1, bolt-1]: Received 8192 of 2..8192 bytes
[conn-1, bolt-1]: S[1] SUCCESS [{}]
[conn-1, bolt-1]: S[2] SUCCESS [{t_first: 97, fields: [n]}]
[conn-1, bolt-1]: <SET result_field_names=[n]>
[conn-1, bolt-1]: S[3] RECORD [[1]]
[conn-1, bolt-1]: S[3] RECORD [[2]]
[conn-1, bolt-1]: S[3] RECORD [[3]]
[conn-1, bolt-1]: Received 7 of 5..8191 bytes
[conn-1, bolt-1]: Received 8192 of 2..8192 bytes
[conn-1, bolt-1]: Received 8 of 6..8192 bytes
[conn-1, bolt-1]: Received 8192 of 2..8192 bytes
[conn-1, bolt-1]: Received 8 of 6..8192 bytes
[conn-1, bolt-1]: Received 8192 of 2..8192 bytes
[conn-1, bolt-1]: Received 8 of 6..8192 bytes
[conn-1, bolt-1]: Received 8192 of 2..8192 bytes
[conn-1, bolt-1]: Received 8 of 6..8192 bytes
[conn-1, bolt-1]: Received 8192 of 2..8192 bytes
[conn-1, bolt-1]: Received 8 of 6..8192 bytes
[conn-1, bolt-1]: Received 8192 of 2..8192 bytes
[conn-1, bolt-1]: Received 8 of 6..8192 bytes
[conn-1, bolt-1]: Received 8192 of 2..8192 bytes
[conn-1, bolt-1]: Received 8 of 6..8192 bytes
[conn-1, bolt-1]: Received 8192 of 2..8192 bytes
[conn-1, bolt-1]: Received 8 of 6..8192 bytes
[conn-1, bolt-1]: Received 8192 of 2..8192 bytes
[conn-1, bolt-1]: Received 8 of 6..8192 bytes
[conn-1, bolt-1]: Received 8192 of 2..8192 bytes
[conn-1, bolt-1]: Received 8 of 6..8192 bytes
[conn-1, bolt-1]: Received 8192 of 2..8192 bytes
[conn-1, bolt-1]: Received 8 of 6..8192 bytes
[conn-1, bolt-1]: Received 1441 of 2..8192 bytes
[conn-1, bolt-1]: S[3]: Received 9997 more records
[conn-1, bolt-1]: S[3] SUCCESS [{type: r, t_last: 63}]
[conn-1, bolt-1]: S[4] SUCCESS [{bookmark: neo4j:bookmark:v1:tx3}]
[conn-1, bolt-1]: <SET last_bookmark="neo4j:bookmark:v1:tx3">
[pool-1]: Releasing connection to pool towards localhost:7687
[conn-1, bolt-1]: C[5] RESET []
[conn-1, bolt-1]: (Sent 6 of 6 bytes)
[conn-1, bolt-1]: Received 7 of 2..8192 bytes
[conn-1, bolt-1]: S[5] SUCCESS [{}]
query                : UNWIND range(1, 10000) AS n RETURN n
record count         : 10000
=====================================
initialisation       : 0s 190484238ns
query transmission   : 0s 000038433ns
query processing     : 0s 131257694ns
result processing    : 0s 039469840ns
shutdown             : 0s 004428413ns
=====================================
TOTAL                : 0s 365678618ns
[pool-1]: Releasing connection to pool towards localhost:7687
[conn-1, bolt-1]: C[6] RESET []
[conn-1, bolt-1]: (Sent 6 of 6 bytes)
[conn-1, bolt-1]: Received 7 of 2..8192 bytes
[conn-1, bolt-1]: S[6] SUCCESS [{}]
[pool-1]: Destroying pool towards localhost:7687
[conn-1, bolt-1]: Connection alive for 0s 370396902ns
[conn-1, bolt-1]: Closing connection
[conn-1, bolt-1]: C[7] GOODBYE []
[conn-1, bolt-1]: (Sent 6 of 6 bytes)
[conn-1]: <DISCONNECTED>
=====================================
current allocation   : 0 bytes
peak allocation      : 49179 bytes
allocation events    : 432
=====================================
-- Running tests
===============================================================================
All tests passed (710 assertions in 47 test cases)

Seabolt test run completed at 2018-11-30T17:58:53
```

But running the suite from the seabolt-ci docker container against the neo4j:latest container, the tests that involve TLS (possibly all?) fail and in fact there is a segfault using the following docker-compose.yml:

```yaml
version: '2'
services:
  neo4j-server:
    image: neo4j:latest
    expose:
    - &bolt-port 7687
    - &http-port 7474
    - &https-port 7473
    ports:
    - 7687:7687
    - 7474:7474
    - 7473:7473
    environment:
      NEO4J_AUTH: neo4j/password
      NEO4J_dbms_connector_bolt_listen__address: ":7687"
      NEO4J_dbms_connector_http_listen__address: ":7474"
      NEO4J_dbms_connector_https_listen__address: ":7473"
      #NEO4J_dbms_connectors_default__listen__address: "::"
    volumes:
    - /tmp/neo4j/conf:/conf

  seabolt-ci:
    depends_on: [neo4j-server]
    command: ./run_tests.sh
    build:
      context: .
      dockerfile: ./ci/linux/Dockerfile.build
    environment:
      BOLT_PORT: 7687
      BOLT_USER: neo4j
      BOLT_HOST: neo4j-server
      PASSWORD: password
      HAVE_SERVER: "yes"
      CHECK_SERVER_SLEEP: 15
```

```shell
seabolt-ci_1_d863accaee29 | Seabolt test run started at 2018-11-30T23:03:41
neo4j-server_1_9e1ceec8baef | Changed password for user 'neo4j'.
neo4j-server_1_9e1ceec8baef | Active database: graph.db
neo4j-server_1_9e1ceec8baef | Directories in use:
neo4j-server_1_9e1ceec8baef |   home:         /var/lib/neo4j
neo4j-server_1_9e1ceec8baef |   config:       /var/lib/neo4j/conf
neo4j-server_1_9e1ceec8baef |   logs:         /var/lib/neo4j/logs
neo4j-server_1_9e1ceec8baef |   plugins:      /var/lib/neo4j/plugins
neo4j-server_1_9e1ceec8baef |   import:       /var/lib/neo4j/import
neo4j-server_1_9e1ceec8baef |   data:         /var/lib/neo4j/data
neo4j-server_1_9e1ceec8baef |   certificates: /var/lib/neo4j/certificates
neo4j-server_1_9e1ceec8baef |   run:          /var/lib/neo4j/run
neo4j-server_1_9e1ceec8baef | Starting Neo4j.
neo4j-server_1_9e1ceec8baef | 2018-11-30 23:03:43.208+0000 WARN  Unknown config option: causal_clustering.discovery_listen_address
neo4j-server_1_9e1ceec8baef | 2018-11-30 23:03:43.210+0000 WARN  Unknown config option: causal_clustering.raft_advertised_address
neo4j-server_1_9e1ceec8baef | 2018-11-30 23:03:43.210+0000 WARN  Unknown config option: causal_clustering.raft_listen_address
neo4j-server_1_9e1ceec8baef | 2018-11-30 23:03:43.211+0000 WARN  Unknown config option: ha.host.coordination
neo4j-server_1_9e1ceec8baef | 2018-11-30 23:03:43.211+0000 WARN  Unknown config option: causal_clustering.transaction_advertised_address
neo4j-server_1_9e1ceec8baef | 2018-11-30 23:03:43.211+0000 WARN  Unknown config option: causal_clustering.discovery_advertised_address
neo4j-server_1_9e1ceec8baef | 2018-11-30 23:03:43.211+0000 WARN  Unknown config option: ha.host.data
neo4j-server_1_9e1ceec8baef | 2018-11-30 23:03:43.211+0000 WARN  Unknown config option: causal_clustering.transaction_listen_address
neo4j-server_1_9e1ceec8baef | 2018-11-30 23:03:43.220+0000 INFO  ======== Neo4j 3.5.0 ========
neo4j-server_1_9e1ceec8baef | 2018-11-30 23:03:43.236+0000 INFO  Starting...
neo4j-server_1_9e1ceec8baef | 2018-11-30 23:03:45.416+0000 INFO  Bolt enabled on neo4j-server:7687.
neo4j-server_1_9e1ceec8baef | 2018-11-30 23:03:46.710+0000 INFO  Started.
neo4j-server_1_9e1ceec8baef | 2018-11-30 23:03:47.626+0000 INFO  Remote interface available at http://localhost:7474/
seabolt-ci_1_d863accaee29 | -- Checking server
seabolt-ci_1_d863accaee29 | -- Attempt #1
seabolt-ci_1_d863accaee29 | [connector]: Version 1.7.1-dev [e6ea1aa]
seabolt-ci_1_d863accaee29 | [pool-1]: Creating pool towards neo4j-server:7687
seabolt-ci_1_d863accaee29 | [pool-1]: Acquiring connection from the pool towards neo4j-server:7687
seabolt-ci_1_d863accaee29 | [addr]: Resolving address neo4j-server:7687
seabolt-ci_1_d863accaee29 | [addr]: Host resolved to 1 IP address
seabolt-ci_1_d863accaee29 | [conn-1]: Opening IPv4 connection to 192.168.224.2 at port 7687
seabolt-ci_1_d863accaee29 | [conn-1]: Securing socket
seabolt-ci_1_d863accaee29 | [pool-1]: Openssl reported error 'self signed certificate' with code '18' when establishing trust, but resuming handshake since trust verification is set to be skipped
seabolt-ci_1_d863accaee29 | [pool-1]: Openssl reported failure of hostname verification due to a mismatch, but resuming handshake since hostname verification is set to be skipped
seabolt-ci_1_d863accaee29 | [pool-1]: Openssl established trust
seabolt-ci_1_d863accaee29 | [conn-1]: Performing handshake
seabolt-ci_1_d863accaee29 | [conn-1]: (Sent 20 of 20 bytes)
seabolt-ci_1_d863accaee29 | [conn-1]: Received 4 of 4 bytes
seabolt-ci_1_d863accaee29 | [conn-1]: <SET protocol_version=3>
seabolt-ci_1_d863accaee29 | [conn-1]: Remote endpoint is 192.168.224.2:7687
seabolt-ci_1_d863accaee29 | [conn-1]: Local endpoint is 192.168.224.3:33038
seabolt-ci_1_d863accaee29 | [conn-1]: <CONNECTED>
seabolt-ci_1_d863accaee29 | [conn-1]: Initialising connection
seabolt-ci_1_d863accaee29 | [conn-1]: C[0] HELLO [{scheme: basic, principal: neo4j, credentials: ********, user_agent: seabolt/1.7.1-dev}]
seabolt-ci_1_d863accaee29 | [conn-1]: (Sent 87 of 87 bytes)
seabolt-ci_1_d863accaee29 | [conn-1]: Received 47 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1]: S[0] SUCCESS [{server: Neo4j/3.5.0, connection_id: bolt-0}]
seabolt-ci_1_d863accaee29 | [conn-1]: <SET server="Neo4j/3.5.0">
seabolt-ci_1_d863accaee29 | [conn-1]: <SET connection_id="bolt-0">
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: <READY>
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: C[1] BEGIN [{}]
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: C[2] RUN [UNWIND range(1, 10000) AS n RETURN n, {}, {}]
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: C[3] PULL_ALL []
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: C[4] COMMIT []
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: (Sent 65 of 65 bytes)
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8192 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: S[1] SUCCESS [{}]
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: S[2] SUCCESS [{t_first: 109, fields: [n]}]
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: <SET result_field_names=[n]>
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: S[3] RECORD [[1]]
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: S[3] RECORD [[2]]
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: S[3] RECORD [[3]]
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 7 of 5..8191 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8192 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8 of 6..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8192 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8 of 6..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8192 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8 of 6..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8192 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8 of 6..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8192 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8 of 6..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8192 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8 of 6..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8192 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8 of 6..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8192 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8 of 6..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8192 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8 of 6..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8192 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8 of 6..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8192 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 8 of 6..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 1441 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: S[3]: Received 9997 more records
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: S[3] SUCCESS [{type: r, t_last: 82}]
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: S[4] SUCCESS [{bookmark: neo4j:bookmark:v1:tx1}]
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: <SET last_bookmark="neo4j:bookmark:v1:tx1">
seabolt-ci_1_d863accaee29 | [pool-1]: Releasing connection to pool towards neo4j-server:7687
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: C[5] RESET []
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: (Sent 6 of 6 bytes)
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 7 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: S[5] SUCCESS [{}]
seabolt-ci_1_d863accaee29 | query                : UNWIND range(1, 10000) AS n RETURN n
seabolt-ci_1_d863accaee29 | record count         : 10000
seabolt-ci_1_d863accaee29 | =====================================
seabolt-ci_1_d863accaee29 | initialisation       : 0s 219644417ns
seabolt-ci_1_d863accaee29 | query transmission   : 0s 000115109ns
seabolt-ci_1_d863accaee29 | query processing     : 0s 138124514ns
seabolt-ci_1_d863accaee29 | result processing    : 0s 061640574ns
seabolt-ci_1_d863accaee29 | shutdown             : 0s 001556195ns
seabolt-ci_1_d863accaee29 | =====================================
seabolt-ci_1_d863accaee29 | TOTAL                : 0s 421080809ns
seabolt-ci_1_d863accaee29 | [pool-1]: Releasing connection to pool towards neo4j-server:7687
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: C[6] RESET []
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: (Sent 6 of 6 bytes)
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Received 7 of 2..8192 bytes
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: S[6] SUCCESS [{}]
seabolt-ci_1_d863accaee29 | [pool-1]: Destroying pool towards neo4j-server:7687
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Connection alive for 0s 421137589ns
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: Closing connection
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: C[7] GOODBYE []
seabolt-ci_1_d863accaee29 | [conn-1, bolt-0]: (Sent 6 of 6 bytes)
seabolt-ci_1_d863accaee29 | [conn-1]: <DISCONNECTED>
seabolt-ci_1_d863accaee29 | =====================================
seabolt-ci_1_d863accaee29 | current allocation   : 0 bytes
seabolt-ci_1_d863accaee29 | peak allocation      : 49080 bytes
seabolt-ci_1_d863accaee29 | allocation events    : 434
seabolt-ci_1_d863accaee29 | =====================================
seabolt-ci_1_d863accaee29 | -- Running tests
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
seabolt-ci_1_d863accaee29 | seabolt-test is a Catch v2.0.1 host application.
seabolt-ci_1_d863accaee29 | Run with -? for options
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test chunking
seabolt-ci_1_d863accaee29 |      Given: an open and initialised connection
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-chunking-v1.cpp:45
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/seabolt.cpp:42: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test basic secure connection (IPv4)
seabolt-ci_1_d863accaee29 |      Given: a local server address
seabolt-ci_1_d863accaee29 |       When: a secure connection is opened
seabolt-ci_1_d863accaee29 |       Then: the connection should be connected
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:32
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:33: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test basic secure connection (IPv6)
seabolt-ci_1_d863accaee29 |      Given: a local server address
seabolt-ci_1_d863accaee29 |       When: a secure connection is opened
seabolt-ci_1_d863accaee29 |       Then: the connection should be connected
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:50
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:51: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test basic insecure connection (IPv4)
seabolt-ci_1_d863accaee29 |      Given: a local server address
seabolt-ci_1_d863accaee29 |       When: an insecure connection is opened
seabolt-ci_1_d863accaee29 |       Then: the connection should be connected
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:68
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:69: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test basic insecure connection (IPv6)
seabolt-ci_1_d863accaee29 |      Given: a local server address
seabolt-ci_1_d863accaee29 |       When: an insecure connection is opened
seabolt-ci_1_d863accaee29 |       Then: the connection should be connected
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:85
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:86: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test connection reuse after graceful shutdown
seabolt-ci_1_d863accaee29 |      Given: a local server address
seabolt-ci_1_d863accaee29 |       When: a secure connection is opened
seabolt-ci_1_d863accaee29 |       Then: the connection should be connected
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:137
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:138: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test connection reuse after graceful shutdown
seabolt-ci_1_d863accaee29 |      Given: a local server address
seabolt-ci_1_d863accaee29 |       When: a secure connection is opened
seabolt-ci_1_d863accaee29 |       Then: the connection should be connected
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:145
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:146: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test connection reuse after graceless shutdown
seabolt-ci_1_d863accaee29 |      Given: a local server address
seabolt-ci_1_d863accaee29 |       When: a secure connection is opened
seabolt-ci_1_d863accaee29 |       Then: the connection should be connected
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:163
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:164: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test connection reuse after graceless shutdown
seabolt-ci_1_d863accaee29 |      Given: a local server address
seabolt-ci_1_d863accaee29 |       When: a secure connection is opened
seabolt-ci_1_d863accaee29 |       Then: the connection should be connected
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:168
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:169: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test init with valid credentials
seabolt-ci_1_d863accaee29 |      Given: an open connection
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:180
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/seabolt.cpp:42: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test init with invalid credentials
seabolt-ci_1_d863accaee29 |      Given: an open connection
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:200
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/seabolt.cpp:42: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test execution of simple Cypher statement
seabolt-ci_1_d863accaee29 |      Given: an open and initialised connection
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:224
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/seabolt.cpp:42: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test field names returned from Cypher execution
seabolt-ci_1_d863accaee29 |      Given: an open and initialised connection
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:245
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/seabolt.cpp:42: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test parameterised Cypher statements
seabolt-ci_1_d863accaee29 |      Given: an open and initialised connection
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:279
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/seabolt.cpp:42: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test execution of multiple Cypher statements transmitted together
seabolt-ci_1_d863accaee29 |      Given: an open and initialised connection
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:310
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/seabolt.cpp:42: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test transactions
seabolt-ci_1_d863accaee29 |      Given: an open and initialised connection
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:333
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/seabolt.cpp:42: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test FAILURE
seabolt-ci_1_d863accaee29 |      Given: an open and initialised connection
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-direct.cpp:382
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/seabolt.cpp:42: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( connection->status->state==1 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 1
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test using a pooled connection
seabolt-ci_1_d863accaee29 |      Given: a new connection pool
seabolt-ci_1_d863accaee29 |       When: a connection is acquired
seabolt-ci_1_d863accaee29 |       Then: the connection should be connected
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-pooling.cpp:36
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-pooling.cpp:37: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( status->state==2 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 2
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test reusing a pooled connection
seabolt-ci_1_d863accaee29 |      Given: a new connection pool with one entry
seabolt-ci_1_d863accaee29 |       When: a connection is acquired, released and acquired again
seabolt-ci_1_d863accaee29 |       Then: the connection should be connected
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-pooling.cpp:66
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-pooling.cpp:67: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( status2->state==2 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 2
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test reusing a pooled connection that was abandoned
seabolt-ci_1_d863accaee29 |      Given: a new connection pool with one entry
seabolt-ci_1_d863accaee29 |       When: a connection is acquired, released and acquired again
seabolt-ci_1_d863accaee29 |       Then: handle should include a valid connection
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-pooling.cpp:97
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-pooling.cpp:98: FAILED:
seabolt-ci_1_d863accaee29 |   REQUIRE( status1->state==2 )
seabolt-ci_1_d863accaee29 | with expansion:
seabolt-ci_1_d863accaee29 |   4 == 2
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | Scenario: Test reusing a pooled connection that was abandoned
seabolt-ci_1_d863accaee29 |      Given: a new connection pool with one entry
seabolt-ci_1_d863accaee29 |       When: a connection is acquired, released and acquired again
seabolt-ci_1_d863accaee29 | -------------------------------------------------------------------------------
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-pooling.cpp:95
seabolt-ci_1_d863accaee29 | ...............................................................................
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | /tmp/seabolt/src/seabolt/tests/test-pooling.cpp:95: FAILED:
seabolt-ci_1_d863accaee29 | due to a fatal error condition:
seabolt-ci_1_d863accaee29 |   SIGSEGV - Segmentation violation signal
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | ===============================================================================
seabolt-ci_1_d863accaee29 | test cases:  25 |   7 passed | 18 failed
seabolt-ci_1_d863accaee29 | assertions: 153 | 132 passed | 21 failed
seabolt-ci_1_d863accaee29 | 
seabolt-ci_1_d863accaee29 | ./run_tests.sh: line 92:    10 Segmentation fault      (core dumped) BOLT_PORT="${BOLT_PORT}" ${BASE}/build/bin/seabolt-test ${TEST_ARGS}
seabolt-ci_1_d863accaee29 | FATAL: Test execution failed.
seabolt_seabolt-ci_1_d863accaee29 exited with code 18
```

The `NEO4J_dbms_connectors_default__listen__address: "::"` is commented out in the above docker-compoose.yml because it produced the following:

```neo4j-server_1_791c18aa74af | 2018-11-30 23:22:52.324+0000 ERROR Failed to start Neo4j: Starting Neo4j failed: Component 'org.neo4j.server.AbstractNeoServer$ServerComponentsLifecycleAdapter@389a1e34' was successfully initialized, but failed to start. Please see the attached cause exception "Protocol family unavailable". Starting Neo4j failed: Component 'org.neo4j.server.AbstractNeoServer$ServerComponentsLifecycleAdapter@389a1e34' was successfully initialized, but failed to start. Please see the attached cause exception "Protocol family unavailable".
neo4j-server_1_791c18aa74af | org.neo4j.server.ServerStartupException: Starting Neo4j failed: Component 'org.neo4j.server.AbstractNeoServer$ServerComponentsLifecycleAdapter@389a1e34' was successfully initialized, but failed to start. Please see the attached cause exception "Protocol family unavailable".
neo4j-server_1_791c18aa74af | 	at org.neo4j.server.exception.ServerStartupErrors.translateToServerStartupError(ServerStartupErrors.java:45)
neo4j-server_1_791c18aa74af | 	at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:184)
neo4j-server_1_791c18aa74af | 	at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:122)
neo4j-server_1_791c18aa74af | 	at org.neo4j.server.ServerBootstrapper.start(ServerBootstrapper.java:90)
neo4j-server_1_791c18aa74af | 	at org.neo4j.server.CommunityEntryPoint.main(CommunityEntryPoint.java:32)
neo4j-server_1_791c18aa74af | Caused by: org.neo4j.kernel.lifecycle.LifecycleException: Component 'org.neo4j.server.AbstractNeoServer$ServerComponentsLifecycleAdapter@389a1e34' was successfully initialized, but failed to start. Please see the attached cause exception "Protocol family unavailable".
neo4j-server_1_791c18aa74af | 	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:473)
neo4j-server_1_791c18aa74af | 	at org.neo4j.kernel.lifecycle.LifeSupport.start(LifeSupport.java:111)
neo4j-server_1_791c18aa74af | 	at org.neo4j.server.AbstractNeoServer.start(AbstractNeoServer.java:177)
neo4j-server_1_791c18aa74af | 	... 3 more
neo4j-server_1_791c18aa74af | Caused by: MultiException[java.net.SocketException: Protocol family unavailable, java.net.SocketException: Protocol family unavailable]
neo4j-server_1_791c18aa74af | 	at org.eclipse.jetty.util.MultiException.ifExceptionThrow(MultiException.java:124)
neo4j-server_1_791c18aa74af | 	at org.eclipse.jetty.server.Server.doStart(Server.java:409)
neo4j-server_1_791c18aa74af | 	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
neo4j-server_1_791c18aa74af | 	at org.neo4j.server.web.Jetty9WebServer.startJetty(Jetty9WebServer.java:330)
neo4j-server_1_791c18aa74af | 	at org.neo4j.server.web.Jetty9WebServer.start(Jetty9WebServer.java:160)
neo4j-server_1_791c18aa74af | 	at org.neo4j.server.AbstractNeoServer.startWebServer(AbstractNeoServer.java:287)
neo4j-server_1_791c18aa74af | 	at org.neo4j.server.AbstractNeoServer.access$900(AbstractNeoServer.java:101)
neo4j-server_1_791c18aa74af | 	at org.neo4j.server.AbstractNeoServer$ServerComponentsLifecycleAdapter.start(AbstractNeoServer.java:530)
neo4j-server_1_791c18aa74af | 	at org.neo4j.kernel.lifecycle.LifeSupport$LifecycleInstance.start(LifeSupport.java:452)
neo4j-server_1_791c18aa74af | 	... 5 more
neo4j-server_1_791c18aa74af | 	Suppressed: java.net.SocketException: Protocol family unavailable
neo4j-server_1_791c18aa74af | 		at sun.nio.ch.Net.bind0(Native Method)
neo4j-server_1_791c18aa74af | 		at sun.nio.ch.Net.bind(Net.java:433)
neo4j-server_1_791c18aa74af | 		at sun.nio.ch.Net.bind(Net.java:425)
neo4j-server_1_791c18aa74af | 		at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:223)
neo4j-server_1_791c18aa74af | 		at sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:74)
neo4j-server_1_791c18aa74af | 		at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:331)
neo4j-server_1_791c18aa74af | 		at org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:299)
neo4j-server_1_791c18aa74af | 		at org.eclipse.jetty.server.AbstractNetworkConnector.doStart(AbstractNetworkConnector.java:80)
neo4j-server_1_791c18aa74af | 		at org.eclipse.jetty.server.ServerConnector.doStart(ServerConnector.java:235)
neo4j-server_1_791c18aa74af | 		at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
neo4j-server_1_791c18aa74af | 		at org.eclipse.jetty.server.Server.doStart(Server.java:398)
neo4j-server_1_791c18aa74af | 		... 12 more
neo4j-server_1_791c18aa74af | 	Suppressed: java.net.SocketException: Protocol family unavailable
neo4j-server_1_791c18aa74af | 		at sun.nio.ch.Net.bind0(Native Method)
neo4j-server_1_791c18aa74af | 		at sun.nio.ch.Net.bind(Net.java:433)
neo4j-server_1_791c18aa74af | 		at sun.nio.ch.Net.bind(Net.java:425)
neo4j-server_1_791c18aa74af | 		at sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:223)
neo4j-server_1_791c18aa74af | 		at sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:74)
neo4j-server_1_791c18aa74af | 		at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:331)
neo4j-server_1_791c18aa74af | 		at org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:299)
neo4j-server_1_791c18aa74af | 		at org.eclipse.jetty.server.AbstractNetworkConnector.doStart(AbstractNetworkConnector.java:80)
neo4j-server_1_791c18aa74af | 		at org.eclipse.jetty.server.ServerConnector.doStart(ServerConnector.java:235)
neo4j-server_1_791c18aa74af | 		at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
neo4j-server_1_791c18aa74af | 		at org.eclipse.jetty.server.Server.doStart(Server.java:398)
neo4j-server_1_791c18aa74af | 		... 12 more
```

So it seems there's quite a bit going wrong here and I'm not really sure what's relevant. So sorry for the big log dump. (Also wish github would let me post these as files... maybe they can.)